src/Hazelcast.Net/Core/ConnectionRetryOptions.cs

```diff
@@ -55,9 +53,9 @@ private RetryOptions(RetryOptions other)
         public double Multiplier { get; set; } = 1;
 
         /// <summary>
-        /// Gets or sets the cluster connection timeout in milliseconds.
+        /// Gets or sets the timeout in milliseconds.
         /// </summary>
-        public long ClusterConnectionTimeoutMilliseconds { get; set; } = 20*1000;
+        public long ConnectionTimeoutMilliseconds { get; set; } = -1; // infinite
```

Comment:
I think the ClusterConnectionTimeoutMiliseconds name (and its documentation) was better than the current state